pkg/provisioner/ironic/testserver/ironic.go

```diff
@@ -76,23 +76,46 @@ func (m *IronicMock) WithDrivers() *IronicMock {
 	return m
 }
 
-// WithNode configures the server with a valid response for /v1/nodes
-func (m *IronicMock) WithNode(node nodes.Node) *IronicMock {
+func (m *IronicMock) buildURL(url string, method string) string {
+	return fmt.Sprintf("%s:%s", url, method)
+}
+
+func (m *IronicMock) withNode(node nodes.Node, method string) *IronicMock {
+
 	if node.UUID != "" {
-		m.ResponseJSON("/v1/nodes/"+node.UUID, node)
+		m.ResponseJSON(m.buildURL("/v1/nodes/"+node.UUID, method), node)
 	}
 	if node.Name != "" {
-		m.ResponseJSON("/v1/nodes/"+node.Name, node)
+		m.ResponseJSON(m.buildURL("/v1/nodes/"+node.Name, method), node)
 	}
 	return m
 }
 
-// WithNodeStatesProvision configures the server with a valid response for /v1/nodes/<node>/states/provision
-func (m *IronicMock) WithNodeStatesProvision(nodeUUID string) *IronicMock {
-	m.ResponseWithCode("/v1/nodes/"+nodeUUID+"/states/provision", "{}", http.StatusAccepted)
+// WithNode configures the server with a valid response for [GET] /v1/nodes
+func (m *IronicMock) WithNode(node nodes.Node) *IronicMock {
+	return m.withNode(node, http.MethodGet)
+}
+
+// WithNodeUpdate configures the server with a valid response for [PATCH] /v1/nodes
```

Comment:
Is it really valid, though? It's going to return the node passed to the method, not one changed by a patch, right?

Reply:
I didn't want to change the previous semantic in this commit, probably we could refine better some of the update methods.

Reply:
So we need this method because some of the existing API calls are not using GET and now the default for `WithNode()` only supports GET?

Reply:
Exactly. I think in this way we could mimic in a more realistic way the ironic/inspector API where for distinct HTTP methods there could be different behaviors (from our point of view, a different json/code response). I also hit the limits of the previous implementation while developing the tests for the Provisioner.Delete() func.

Reply:
OK, the concern I have is that we're not actually testing that the update does anything. That's what I was trying to do in #690 , although I can see there being a concern with re-implementing the patch logic there if it gets much more complicated.

What do you see for the future of this method? Would I rebase 690 on these changes and replace the implementation of this method with something more dynamic?

Reply:
I think it would be really fine. I had to provide a basic backward compatible implementation for it due the infrastructural changes to support the HTTP methods dispatching, otherwise the tests failed due a missing handler (for the PATCH).
So, the current implementation for NodeUpdate does not change the previous behavior, and it will allow to merge this PR and then rework it as needed.

Reply:
Sure, I understand. I'm trying to work out what the next step might be, after we take this patch. I think I see what that would be, now.